src/platform_impl/linux/wayland/window/mod.rs

```diff
@@ -105,37 +106,79 @@ impl Window {
             .map(|size| size.to_logical::<u32>(1.))
             .unwrap_or((800, 600).into());
 
-        let window = state.xdg_shell.create_window(
-            surface.clone(),
-            WindowDecorations::ServerDefault,
-            &queue_handle,
-        );
-
-        let mut window_state = WindowState::new(
-            event_loop_window_target.connection.clone(),
-            &event_loop_window_target.queue_handle,
-            &state,
-            size,
-            window.clone(),
-            attributes.preferred_theme,
-        );
-
-        // Set the app_id.
-        if let Some(name) = platform_attributes.name.map(|name| name.general) {
-            window.set_app_id(name);
-        }
+        let (window, window_state) = if let Some(layer) = platform_attributes.layer_shell {
+            let layer_surface = state.layer_shell.create_layer_surface(
+                &queue_handle,
+                surface.clone(),
+                layer,
+                // TODO(theonlymrcat): Is this where app id should go?
```

Comment:
Layer shell surfaces don't have app ids. They have a `namespace`, which unlike app ids doesn't have any defined meaning or use.

Reply:
That's true, but I was wondering whether `with_name` should be used to define the `namespace`.